src/util/usePreLoader.ts

```diff
@@ -0,0 +1,45 @@
+import {Story} from "../interfaces";
+import {useEffect, useState} from "react";
+
+
+// Caches given Story[] using HTMLImageElement and HTMLVideoElement
+const cacheContent = async (contents: Story[]) => {
+	const promises = contents.map((content) => {
+		return new Promise(function (resolve, reject) {
+			if(!content.url) return
+
+			if(content.type === 'video') {
```

Comment:
Video preloading is a bit involved[1] and not this direct so I would not do it for now. Also, it might drastically increase data usage when the story may never play if the user scrolls past. Let's do only images, to begin with.

[1]: https://web.dev/fast-playback-with-preload/#manual-buffering

Reply:
Looking the network-profiler + example-app again, videos does seem to preload fine. But I understand the concern of having uncontrolled large video content loading behind the scenes.

But I can also see cases for smaller video clips (as I'm working on myself) where this would be a nice to have.

How about disabling it by default, and allowing opt-in using a `preLoadVideosExperimental` prop?

Reply:
Sounds good! You can even keep this at the `story` object level so that users may choose to individually preload short and long videos. So something like
```ts
{
  type: 'video',
  url: '...',
  preloadResource: true // this may also be used for enabling/disabling at per image level
}
```

What do you think?

Reply:
Yeah sounds good! So images preloads by default, videos does not (unless preLoadVideosExperimental is set? [do we still want this?] ). 

Further, a `preloadResource: boolean` can be set on `Story` to overwrite on object level.

Reply:
We don't need `preLoadVideosExperimental` if it is done on a story level.